libshaderc_spvc/src/spvc.cc

```diff
@@ -171,7 +202,88 @@ shaderc_spvc_compilation_result_t validate_and_compile(
 shaderc_spvc_compilation_result_t shaderc_spvc_compile_into_glsl(
     const shaderc_spvc_compiler_t, const uint32_t* source, size_t source_len,
     shaderc_spvc_compile_options_t options) {
+  auto* result = new (std::nothrow) shaderc_spvc_compilation_result;
+  if (!result) return nullptr;
+
   spirv_cross::CompilerGLSL compiler(source, source_len);
+
+  if (options->glsl.version == 0) {
+    // no version requested, was one detected in source?
+    options->glsl.version = compiler.get_common_options().version;
+    if (options->glsl.version == 0) {
+      // no version detected in source, use default
+      options->glsl.version = 450;
```

Comment:
450 should probably be a constant somewhere

Reply:
will do